### ISSUE TITLE
All downloaded files are same size as Content-Length header value

### ISSUE THREAD

**Issue by metafive** [andreafabrizi/Dropbox-Uploader#245]
On Debian, with -d I get the files to download partially.  They are even intact .gz files, though shortened, as I can gunzip filename and it produces a nice ascii file I can look at w/ text editor.  So, files smaller than 587070 bytes show their true size after download, but others that are typically in the 20 to 30 mb range cut off at 587070 bytes in the Debian file system.  I have a paid dropbox account, but not a business level one.  Here is the debug file, if this helps at all.
-----------------------------------------------
.../bin# more /tmp/du_resp_debug
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 28 Sep 2015 18:38:56 GMT
Content-Type: application/octet-stream
Content-Length: 587070
Connection: keep-alive
content-security-policy: sandbox allow-scripts allow-forms allow-popups allow-top-navigation; referrer no-referrer;
accept-ranges: bytes
x-dropbox-metadata: {"rev": "903b501053", "thumb_exists": false, "path": "/Apps/UpdraftPlus/backup_2015-09-26-0233_ChangedName_Services_96828a6ac825-
db.gz", "is_dir": false, "client_mtime": "Sat, 26 Sep 2015 02:40:22 +0000", "icon": "page_white_compressed", "read_only": false, "modifier": null,
"bytes": 587070, "modified": "Sat, 26 Sep 2015 02:40:22 +0000", "size": "573.3 KB", "root": "dropbox", "mime_type": "application/octet-stream", "re
vision": 144}
x-server-response-time: 284
content-security-policy-report-only: default-src *; script-src 'none'; report-uri https://www.dropbox.com/csp_log
etag: 144n
x-dropbox-request-id: 38f7---obfuscated---22c4
pragma: public
cache-control: max-age=0
x-content-security-policy: sandbox allow-scripts allow-forms allow-popups allow-top-navigation; referrer no-referrer;
x-webkit-csp: sandbox allow-scripts allow-forms allow-popups allow-top-navigation; referrer no-referrer;
X-Robots-Tag: noindex, nofollow, noimageindex
-----------------------------------------------

**Comment by andreafabrizi** [comment]
@metafive, sorry I don't understand what's the problem

**Comment by metafive** [comment]
I am referring to the file sizes on the Linux server of the files that are downloaded from dropbox, using the dropbox_uploader.sh.  Here is what I see in Linux:

.../some-user$ ls -al
drwxr-xr-x  2 root root    4096 Sep 29 08:54 .
drwxr-xr-x 28 root root    4096 Sep 28 12:14 ..
-rw-------  1 root root  587070 Sep 28 11:00 825-db.gz
-rw-------  1 root root  587070 Sep 28 11:02 825-plugins.zip
-rw-------  1 root root  587070 Sep 28 11:04 825-plugins2.zip
-rw-------  1 root root  587070 Sep 28 11:07 825-plugins3.zip
-rw-------  1 root root  587070 Sep 28 11:38 825-plugins4.zip
-rw-------  1 root root  587070 Sep 28 11:01 others.zip

Notice how the file size on disk in Linux happens to be the same size as the Content-Length header as found in the debug file that dropbox_uploader.sh produces with the -d option:
"Content-Length: 587070"
But, the files in dropbox are often over 20 mb.

Here is the command I used to download one of the files:
./dropbox_uploader.sh -d download Apps/UpdraftPlus/backup_2015-09-26-0233_Home_Services_96828a6ac825-db.gz ../wpmi/custs/anonymous/825-db.zip

**Comment by metafive** [comment]
RESOLVED: I resolved this by clicking on other directories on the dropbox server via its browser interface and downloading another large file, which worked.  After doing that, I went back into the directory on dropbox server where I copied the filename to put back into your tool for another download w/ same syntax.  This time it downloaded a 20mb file, no problem.  This can be my workaround, I guess, if I encounter this again.  I assume it is some kind of caching bug with dropbox and nginx.
